pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -227,14 +227,40 @@ type FlavorAssignment struct {
 // be assigned immediately. Each assigned flavor is accompanied with a
 // FlavorAssignmentMode.
 func AssignFlavors(log logr.Logger, wl *workload.Info, resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, cq *cache.ClusterQueue) Assignment {
+	fullAssignment := assignFlavors(log, wl.TotalRequests, wl.Obj.Spec.PodSets, resourceFlavors, cq)
+	if fullAssignment.RepresentativeMode() == NoFit {
```

Comment:
what if the RepresentativeMode is Preempt, but the preemption algorithm can't find any candidates for preemption?

We should still attempt a resize in that case.

Here is where it might be useful to do so in a next scheduling attempt, to keep the overall scheduling cycle simple.

Reply:
I've missed that when changing from "partial before preemption", Still I'd prefer to keep it all in a schedule cycle add run the preemption nomination part in the flavor assignment.